pilot/pkg/networking/plugin/authn/util.go

```diff
@@ -0,0 +1,28 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authn
+
+import (
+	"istio.io/istio/pilot/pkg/features"
+	"istio.io/istio/pkg/spiffe"
+)
+
+func trustDomainsForValidation(trustDomainAliases []string) []string {
+	var trustDomains []string
+	if !features.SkipValidateTrustDomain.Get() {
+		trustDomains = append(trustDomainAliases, spiffe.GetTrustDomain())
```

Comment:
1. Please stop using spifee.GetTrustDomain() ( even remove it) - pass  MeshConfig as parameter.
2. I assume this is only for Sidecar ? For Gateways we must add the SAN list explicitly defined by user.

Reply:
1. @yangminzhu is it okay if we revert back to my initial proposal of passing in meshconfig and not the string list?
2. Yes it's only for Sidecar.

Reply:
can we just change trustDomainsForValidation to take the meshconfig as input? This function can just be reused in gateway code path later.